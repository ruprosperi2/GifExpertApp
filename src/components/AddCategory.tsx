import {ChangeEvent, Dispatch, FormEvent, SetStateAction, useState} from "react";

const AddCategory = ({setCategories}: {
    setCategories: Dispatch<SetStateAction<string[]>>
}) => {
    const [inputValue, setInputValue] = useState('')

    const handleSubmit = (e: FormEvent) => {
        e.preventDefault()

        if (inputValue.trim().length >= 2) {
            setCategories((categories) => [inputValue, ...categories])
        }

        setInputValue('');
    }

    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
        setInputValue(e.currentTarget.value)
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                placeholder={"ingrese la categorÃ­a"}
                onChange={handleChange}
                type="text"
                value={inputValue}
            />
        </form>
    )
}

export default AddCategory