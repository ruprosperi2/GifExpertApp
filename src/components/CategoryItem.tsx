const CategoryItem = ({id, title, url}: { id: string, title: string, url: string }) => {
    return(
        <div>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

export default CategoryItem