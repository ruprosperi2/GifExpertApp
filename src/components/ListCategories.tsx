import CategoryItem from "./CategoryItem";
import {useFetchGifs} from "../hooks/useFetchGifs";

const ListCategories = ({category}: { category: string }) => {

    const {data, loading} = useFetchGifs(category)

    return (
        <>
            <h2>{category}</h2>

            {loading ? 'Cargando' :''}

              <div className={"card-grid"}>
                {data.map((image) => {
                    return (
                        <CategoryItem
                            key={image.id}
                            {...image}
                            />)
                })}
              </div>
        </>
    )
}

export default ListCategories