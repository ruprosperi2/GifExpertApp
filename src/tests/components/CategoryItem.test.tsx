import {describe, expect, test} from "vitest";
import {shallow} from "enzyme";
import CategoryItem from "../../components/CategoryItem";
import {Image} from "../../components/types";

describe("DescripciÃ³n", () => {
    const title = "custom-title"
    const url = "custom-url"
    const image: Image = {
        id: "custom-id",
        title: title,
        url: url,
    }

    test("otra descripcion", () => {
        const wrapper = shallow(<CategoryItem {...image} />)

        expect(wrapper).toMatchSnapshot()
    })
})