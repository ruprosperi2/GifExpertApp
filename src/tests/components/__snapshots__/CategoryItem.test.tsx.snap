// Vitest Snapshot v1

exports[`DescripciÃ³n > otra descripcion 1`] = `
<div>
  <img />
  <p />
</div>
`;
